packages/mobile/src/identity/contactMapping.ts

```diff
@@ -259,13 +257,24 @@ function* fetchWalletAddresses(e164Number: string) {
 
 // Returns a list of account addresses for the identifier received.
 export function* lookupAccountAddressesForIdentifier(id: string) {
+  const channel = yield call(lostAccountsChannel)
```

Comment:
This is creating a channel to read continuously from Firebase that's never closed. It will also completely block this flow if there are no lost accounts or the request fails.

Behind the scenes this is using `.on('value', callback)` which creates the stream. Instead, we could use `.once('value')` which returns a Promise with the result. You can see an example [here](https://github.com/celo-org/wallet/blob/main/packages/mobile/src/consumerIncentives/contentFetcher.ts#L17)